### PR TITLE
SEQNG-879: Nifs CC setup

### PR DESCRIPTION
**Pull request by cquiroz**
This is the code to configure NIFS CC. It is more or less a direct translation of the tcl code though the two pass setup of the CC is now made explicit

This of course requires intensive testing to be verified

### CODE REVIEW

**Comment by cquiroz** [comment]
I see, that explains some details I didn't understand in the code

File: modules/seqexec/server/src/main/scala/seqexec/server/nifs/NifsControllerEpics.scala [hunk 1]
```diff
@@ -96,9 +98,23 @@ trait NifsEncoders {
 object NifsControllerEpics extends NifsController[IO] with NifsEncoders {
   private val Log = getLogger
 
+  private val ConfigTimeout: Time  = Seconds(180)
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/nifs/NifsControllerEpics.scala]: Tcl Seqexec uses 400[s] (NifsInts.itcl:83)

File: modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala [hunk 2]
```diff
@@ -339,8 +339,14 @@ object EpicsUtil {
...
-  def smartSetParamF[F[_]: Monad, A: Eq](v: A, get: => F[Option[A]], set: F[Unit]): F[Unit] =
-    get.map(_ =!= v.some).ifM(set, Applicative[F].unit)
+  // The return signature indicates this programs calculates if we maybe need an action
+  // e.g. it checks that a value in epics compares to a reference and if so returns an optional
+  // action
+  def smartSetParamF[F[_]: Monad, A: Eq](v: A, get: => F[Option[A]], set: F[Unit]): F[Option[F[Unit]]] =
+    get.map(_ =!= v.some).map {
+      case true => set.some
+      case false => none
+    }
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: `(_ =!= v.some).option(set)` ?

Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: `get.map(_ =!=.some)` gives `F[Boolean]`

Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: `get.map((_ =!= v.some).option(set))`

Comment (cquiroz) [reply modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: I think you need an extra map, but yeah that will work

**Pull request merged**